### PR TITLE
numpy 2.0

### LINKED ISSUE #389: pytables incompatible with numpy 2.0
**Describe the bug**
pytables, which is required for hdf5 functionality, is incompatible with numpy 2.0

This demonstrates that, even though numpy and scipy tend to be more stable than matplotlib or pandas, we shouldn't expect compatibility to persist though major version changes.

### PR DESCRIPTION
**Pull request by AdamOrmondroyd**
# Description

Changes for compatibility with numpy 2.0

- `np.trapz` has been deprecated in favour of `np.trapezoid`
- (py)tables is broken https://github.com/PyTables/PyTables/issues/1172

fixes #389 
*TODO: change numpy restriction once #390 is merged and pytables is updated

# Checklist:

- [x] I have performed a self-review of my own code
- [x] My code is PEP8 compliant (`flake8 anesthetic tests`)
- [x] My code contains compliant docstrings (`pydocstyle --convention=numpy anesthetic`)
- [x] New and existing unit tests pass locally with my changes (`python -m pytest`)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have appropriately incremented the [semantic version number](https://semver.org/) in both README.rst and anesthetic/_version.py

### CODE REVIEW

**Comment by AdamOrmondroyd** [comment]
Provided numpy<2.0.0, hdf is working fine on my mac.

**Comment by lukashergt** [comment]
Thanks for this @AdamOrmondroyd! 
I've restricted the CI sphinx check to numpy 1 for now.
What keeps this in draft mode?
How do we want to handle the lacking code coverage (which is coming from us trying to stay backwards compatible with numpy 1)? Should we add a single test suite that uses numpy 1?

**Comment by AdamOrmondroyd** [comment]
> Thanks for this @AdamOrmondroyd! I've restricted the CI sphinx check to numpy 1 for now. 
I think the test suite may have misled you. All the pytests involving fastkde skip rather than fail due to the failed import, which is a flaw in how our optional dependency tests work. Really, the "true" tests should fail.
> What keeps this in draft mode?
I was waiting for fastkde to release a pupil update.
>How do we want to handle the lacking code coverage (which is coming from us trying to stay backwards compatible with numpy 1)? Should we add a single test suite that uses numpy 1?
This should be covered by the minimum dependencies test, if it isn't that's a bug.

**Comment by williamjameshandley** [comment]
I agree that we should support numpy 1 -- given that it was non-zero work for @AdamOrmondroyd to upgrade our code, we shouldn't expect others to. Our test suite should check both cases, but we'll only need one numpy v1 case rather than the full grid

**Comment by AdamOrmondroyd** [comment]
Ah I see the problem - min_requirements doesn't do anything with <= requirements, should we define a minimum numpy and scipy?

**Comment by lukashergt** [comment]
> I think the test suite may have misled you. All the pytests involving fastkde skip rather than fail due to the failed import, which is a flaw in how our optional dependency tests work. Really, the "true" tests should fail.

oh yeah, that is annoying... :/

**Comment by AdamOrmondroyd** [comment]
Do you have a favourite minimum scipy?

**Comment by AdamOrmondroyd** [comment]
Does some dark codecov magic need to happen?

**Comment by lukashergt** [comment]
> Do you have a favourite minimum scipy?

Nope... Leave it for now?

> Does some dark codecov magic need to happen?

Ah cr**, yeah, and I always forget what the right order of things to do is...

File: tests/utils.py [hunk 1]
```diff
@@ -22,7 +22,7 @@ def skipif_no_astropy(param):
 except ImportError:
     pass
 reason = "requires fastkde package"
-condition = 'fastkde' not in sys.modules
+condition = find_spec('fastkde') is None
 raises = ImportError
 fastkde_mark_skip = pytest.mark.skipif(condition, reason=reason)
 fastkde_mark_xfail = pytest.mark.xfail(condition, raises=raises, reason=reason)
```
[the remark this answers is not in the document]
Comment (AdamOrmondroyd) [reply tests/utils.py:28]: Just what we need, thanks!

[the remark this answers is not in the document]
Comment (AdamOrmondroyd) [reply tests/utils.py:28]: works locally with the current main branch of fastkde, just need them to update it on pypi when they've fixed their ci

**Review by lukashergt — APPROVED** [review]
Fantastic, many thanks @AdamOrmondroyd!

**Pull request merged**